packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts

```diff
@@ -32,83 +35,31 @@ export const reactWrapper = <T extends TemplateRenderProps>(
       ? scopedTemplateFilepath
       : rootTemplateFilepath;
 
-  return `<!DOCTYPE html>
-    <html lang=${lang}>
-    <head>
-        <script>window.__INITIAL__DATA__ = ${JSON.stringify(props)}</script>
-        ${Array.from(
-          getCssTags(
-            templateFilepath,
-            props.__meta.manifest.bundlerManifest,
-            new Set()
-          )
-        )
-          .map(
-            (f) =>
-              `<link rel="stylesheet" href="${
-                props.relativePrefixToRoot + f
-              }"/>`
-          )
-          .join("\n")}
-        ${headConfig ? renderHeadConfigToString(headConfig) : ""}
-    </head>
-    <body>
-        <div id="reactele">${template}</div>${
-    hydrate
-      ? `<script type="module" src="${
-          props.relativePrefixToRoot +
-          findHydrationFilename(
-            `${projectFilepaths.hydrationBundleOutputRoot}/${templateModuleInternal.templateName}.tsx`,
-            props
-          )
-        }" defer></script>`
-      : ""
-  }
-    </body>
-    </html>`;
-};
+  const relativeServerRenderPath = serverRenderPath.replace("assets", "..");
+  const serverRenderTemplateModule = (await import(
+    relativeServerRenderPath
+  )) as RenderTemplate;
 
-type chunkName = string;
-type bundlerManifest = Record<chunkName, ManifestInfo>;
+  const serverHtml = await serverRenderTemplateModule.render({
+    Page: templateModuleInternal.default!,
+    pageProps: props,
+  });
 
-const getCssTags = (
-  filepath: string,
-  manifest: bundlerManifest,
-  seen: Set<string>
-): Set<string> => {
-  const entry = structuredClone(
-    Object.entries(manifest).find(([file]) => file === filepath)
+  const clientHtml = getHydrationTemplate(
```

Comment:
might be nice to do this up-front  and just pass the hydration template into this function? not really a big deal either way..

Reply:
I think this is effectively the same comment as the previous one to load the render templates in renderer.ts and pass them through. Yea?

Reply:
yeah.